ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/util/PendingPool.java

```diff
@@ -95,32 +109,39 @@
   public static PendingPool<ValidateableAttestation> createForAttestations() {
     return new PendingPool<>(
         DEFAULT_HISTORICAL_SLOT_TOLERANCE,
-        DEFAULT_FUTURE_SLOT_TOLERANCE,
+        FutureItems.DEFAULT_FUTURE_SLOT_TOLERANCE,
+        DEFAULT_MAX_ITEMS,
         ValidateableAttestation::hash_tree_root,
         ValidateableAttestation::getDependentBlockRoots,
         ValidateableAttestation::getEarliestSlotForForkChoiceProcessing);
   }
 
-  public void add(T item) {
+  public synchronized void add(T item) {
     if (shouldIgnoreItem(item)) {
       // Ignore items outside of the range we care about
       return;
     }
 
+    // Make room for the new item
+    while (pendingItems.size() > (maxItems - 1)) {
+      final SlotAndRoot toRemove = orderedPendingItems.pollFirst();
+      if (toRemove == null) {
```

Comment:
if the while clause is true and the to remove item is null, doesn't that mean the max items is less than 1? It seems a bit redundant, or we might want to just throw an error in this scenario to show that max items shouldn't have been set to 1. Or am I missing something?